app/frontend/wallet/shelley-wallet.ts

```diff
@@ -260,14 +265,28 @@ const ShelleyWallet = ({
     }
   }
 
-  function prepareTxAux(plan) {
+  async function calculateTtl() {
+    return (
+      cryptoProvider.network.ttl +
+      (await blockchainExplorer
+        .getBestSlot()
+        .then((res) => res.Right.bestSlot)
+        .catch(
+          () =>
+            cryptoProvider.network.eraStartSlot +
+            Math.floor((Date.now() - cryptoProvider.network.eraStartDateTime) / 1000)
```

Comment:
This may be the case where I would prefer to use normal try-catch syntax and try not to write all the logic into the return statement. I don't mind that much, but I can imagine this being a bit more readable.

Reply:
Refactored